services/NotificationAgent/src/notification_agent/core.clj

```diff
@@ -4,105 +4,119 @@
         [clojure-commons.lcase-params :only [wrap-lcase-params]]
         [clojure-commons.query-params :only [wrap-query-params]]
         [service-logging.middleware :only [wrap-logging add-user-to-context]]
+        [compojure.api.middleware :only [wrap-exceptions]]
         [compojure.core]
         [korma.db :only [transaction]]
         [ring.middleware keyword-params nested-params]
         [notification-agent.delete]
         [notification-agent.notifications]
         [notification-agent.query]
+        [notification-agent.common :only [success-resp]]
         [notification-agent.seen]
-        [slingshot.slingshot :only [try+]])
+        [slingshot.slingshot :only [try+]]
+        [ring.util.http-response :only [ok]]
+        [ring.util.response :only [content-type]])
   (:require [compojure.route :as route]
             [clojure.tools.logging :as log]
-            [clojure-commons.error-codes :as ce]
+            [clojure-commons.exception :as cx]
             [notification-agent.config :as config]
             [notification-agent.db :as db]
             [ring.adapter.jetty :as jetty]
             [common-cli.core :as ccli]
             [me.raynes.fs :as fs]
-            [service-logging.thread-context :as tc]))
+            [service-logging.thread-context :as tc]
+            [cheshire.core :as cheshire]))
 
-(defn- trap
-  [ctx f]
-  (transaction (ce/trap ctx f)))
+(defn- success-resp
+  "Returns an empty success response."
+  ([]
+   (success-resp nil))
+  ([m]
+   (-> (ok (if (map? m) (cheshire/encode (assoc m :success true)) (str m)))
+       (content-type "application/json; charset=utf-8"))))
+
+(defn- successful-seen-response
+  "Returns the response for a successful request to mark messages seen."
+  [user]
+  (success-resp {:count (str (db/count-matching-messages user {:seen false}))}))
```

Comment:
I'm not sure about moving this kind of logic into the `route` level. Sure, it has `success` and `response` in the name, but it's pulling info from the db and formatting a map that seems specific to `notification-agent.seen` results.
I might leave it in `notification-agent.seen` (maybe rename it without the `successful`) and just wrap the route function calls in `success-resp` or `ok`.